### PR TITLE
p2p libp2p connection send envelope queue

### PR DESCRIPTION
**Pull request by solarw**
## Proposed changes

p2p libp2p connection send envelope queue

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes and CI passes too
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/fetchai/connections/p2p_libp2p/connection.py [hunk 1]
```diff
@@ -770,26 +790,77 @@ async def receive(self, *args: Any, **kwargs: Any) -> Optional["Envelope"]:
...
+        except Exception as e:  # pylint: disable=broad-except
+            self.logger.exception(
+                f"Failed to send. Exception: {e}. Try recover connection to node and send again."
+            )
+
+        try:
+            if self.node.is_proccess_running():
+                await self.node.pipe.connect()
+                await self._node_client.send_envelope(envelope)
+                self.logger.info("Envelope sent after reconnect to node")
```
Comment (DavidMinarsch) [inline packages/fetchai/connections/p2p_libp2p/connection.py:814]: please make it debug level

File: packages/fetchai/connections/p2p_libp2p/connection.py [hunk 2]
```diff
@@ -770,26 +790,77 @@ async def receive(self, *args: Any, **kwargs: Any) -> Optional["Envelope"]:
...
+                return
         except asyncio.CancelledError:  # pylint: disable=try-except-raise
             raise  # pragma: nocover
         except Exception as e:  # pylint: disable=broad-except
+            self.logger.info("Envelope sending failed after reconnect to node")
             self.logger.exception(
-                f"Failed to send. Exception: {e}. Try reconnect to node and read again."
+                f"Failed to send after pipe reconnect. Exception: {e}. Try recover connection to node and send again."
             )
```
Comment (DavidMinarsch) [inline packages/fetchai/connections/p2p_libp2p/connection.py:822]: One log level - exception - is enough.

**Review by DavidMinarsch — APPROVED** [review]

File: packages/fetchai/connections/p2p_libp2p/connection.py (hunk 1 above)
Comment (solarw) [reply packages/fetchai/connections/p2p_libp2p/connection.py]: done

File: packages/fetchai/connections/p2p_libp2p/connection.py (hunk 2 above)
Comment (solarw) [reply packages/fetchai/connections/p2p_libp2p/connection.py]: done

**Pull request merged**